### PR TITLE
Adding iOS App Support

### PR DESCRIPTION
**Pull request by sparsh3dwe**
- Adding iOS app support for OpenBot.
- Written in swift.
- Dual theme (Light/Dark).
- Supports Mobile Controller as well as Remote Controller with MFI enabled.

### CODE REVIEW

**Comment by thias15** [comment]
Hi @sparsh3dwe. There seems to be a lot of files such as `xcuserdata` that don't need to be there. Could you please check which files are build/temporary files not needed for compilation and remove them. We should only have the necessary source code.

File: .gitignore [hunk 1]
```diff
@@ -6,6 +6,10 @@ composer.lock
 .DS_Store
 Thumbs.db
 wandb/
+/ios/OpenBot/Pods/
```
Comment (thias15) [inline .gitignore:9]: There should be another .gitignore file at the level `ios/OpenBot` that has `Pods`, etc. in it. Does not make sense to put this at the root level.

File: firmware/openbot_nano/openbot_nano.ino [hunk 2]
```diff
@@ -116,7 +118,7 @@ const int PIN_PWM_R1 = 9;
 const int PIN_PWM_R2 = 10;
 const int PIN_SPEED_LF = 2;
 const int PIN_SPEED_RF = 3;
-const int PIN_VIN = A7;
+//const int PIN_VIN = A7;
```
Comment (thias15) [inline firmware/openbot_nano/openbot_nano.ino:121]: Why is this disabled? The changes here should not be part of the PR.

File: firmware/openbot_ble/openbot_ble.ino [hunk 3]
```diff
@@ -0,0 +1,1856 @@
+// Required App Version: 0.5
```
Comment (thias15) [inline firmware/openbot_ble/openbot_ble.ino:1]: How many changes are required to the firmware? I think it would make more sense to integrate it with the existing firmware and use a flag such as #HAS_BLUETOOTH to turn the feature on/off in the firmware.

**Comment by thias15** [comment]
`ios/OpenBot/OpenBot.xcodeproj/project.xcworkspace/xcuserdata` is in .gitignore but files like `nitishyadav.xcuserdatad/UserInterfaceState.xcuserstate` are still included in this branch.

File: firmware/openbot_ble/openbot_ble.ino (hunk 3 above)
Comment (sparsh3dwe) [reply firmware/openbot_ble/openbot_ble.ino]: Hi, I've merged the both BLE with the BLUETOOTH flag.

**Comment by thias15** [comment]
@quentin-leboutet how is the testing going?
